MonitoringTools/usr/local/bin/collect_vm_stats.py

```diff
@@ -0,0 +1,121 @@
+import openstack
+
+
+def number_servers_total(conn):
+    """
+    Query an OpenStack Cloud to find the total number of instances across
+    all projects.
+    :param conn: OpenStack cloud connection
+    :returns: Number of VMs in total across the cloud
+    """
+    server_obj = conn.compute.servers(details=False, all_projects=True, limit=10000)
+    # get number of items in generator object
+    instance_list = list(server_obj)
+    total_instances = len(instance_list)
+    return total_instances
+
+
+def number_servers_active(conn):
+    """
+    Query an OpenStack Cloud to find the number of instances in
+    ACTIVE state.
+    :param conn: OpenStack Cloud Connection
+    :returns: Number of active VMs
+    """
+    server_obj = conn.compute.servers(
+        details=False, all_projects=True, limit=10000, status="ACTIVE"
+    )
+    # get number of items in generator object
+    instance_list = list(server_obj)
+    instance_active = len(instance_list)
+    return instance_active
+
+
+def number_servers_build(conn):
+    """
+    Query an OpenStack Cloud to find the number of instances in
+    BUILD state.
+    :param conn: OpenStack Cloud Connection
+    :returns: Number of VMs in BUILD state
+    """
+    server_obj = conn.compute.servers(
+        details=False, all_projects=True, limit=10000, status="BUILD"
+    )
+    # get number of items in generator object
+    instance_list = list(server_obj)
+    instance_build = len(instance_list)
+    return instance_build
+
+
+def number_servers_error(conn):
+    """
+    Query an OpenStack Cloud to find the number of instances in
+    ERROR state.
+    :param conn: OpenStack Cloud Connection
+    :returns: Number of VMs in ERROR state
+    """
+    server_obj = conn.compute.servers(
+        details=False, all_projects=True, limit=10000, status="ERROR"
+    )
+    # get number of items in generator object
+    instance_list = list(server_obj)
+    instance_err = len(instance_list)
+    return instance_err
+
+
+def number_servers_shutoff(conn):
+    """
+    Query an OpenStack Cloud to find the number of instances in
+    SHUTOFF state.
+    :param conn: OpenStack Cloud Connection
+    :returns: Number of VMs in SHUTOFF (STOPPED) state
+    """
+    server_obj = conn.compute.servers(
+        details=False, all_projects=True, limit=10000, status="SHUTOFF"
+    )
+    # get number of items in generator object
+    instance_list = list(server_obj)
+    instance_shutoff = len(instance_list)
+    return instance_shutoff
+
+
+def collect_stats(cloud, prod):
```

Comment:
NIT: make name clearer
```suggestion
def get_all_server_statuses(cloud, prod):
```